lib/utils/pagination.js

```diff
@@ -0,0 +1,46 @@
+function pageNumber(number, size, recordCount) {
+  let numberOfPages = Math.ceil(recordCount / size);
+
+  number = parseInt(number) || 1;
+
+  // Ensure page number is in range
+  number = number < 1 ? 1 : number;
+  number = number > numberOfPages ? numberOfPages : number;
+
+  return number;
+}
+
+function buildIds(order, number, size, recordCount) {
+  let offset = size * (number - 1);
+
+  let start;
+  let mapFunction;
+
+  if (order === 'asc') {
+    start = 1 + offset;
+    mapFunction = (_, i) => start + i;
+  } else {
+    start = recordCount - offset;
+    mapFunction = (_, i) => start - i;
+  }
+
+  // Ensure size is in range
+  let end = offset + size;
+  if (end > recordCount) {
+    let diff = end - recordCount;
+    size = size - diff;
+  }
+
+  return Array.from({ length: size }, mapFunction);
+}
+
+module.exports = function paged(recordCount, options = {}) {
```

Comment:
the call would be somthing like: 
`page(100, { order: 'desc', page: { size: 10, number: 1} })` ?

was wondering if that is less verbose? 
`page(100, { order: 'desc', per_page: 10, page: 1 })`  - but that's probably my ruby will_paginate head :D

Reply:
Yeah, I'm not sure which one is better. I took it from [JSON:API spec](https://jsonapi.org/format/#fetching-pagination) but I don't care how we do it.
@skddc what is your preference?

Reply:
ah ok, then I guess JSON:API spec is better!